pkg/version/semver/semver.go

```diff
@@ -0,0 +1,47 @@
+package semver
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+var (
+	releasePattern       = regexp.MustCompile("^v[0-9]")
+	branchReleasePattern = regexp.MustCompile("^v[0-9].x")
+)
+
+type Version string
+
+// HasReleasePrefix validates the value against the "legacy" releasePattern
+// It is not necessarily a valid SemVer, but does have the expected "release prefix".
+func (v Version) HasReleasePrefix() bool {
+	return releasePattern.MatchString(string(v))
+}
+
+// HasBranchPrefix validates the value as having a minor version set to `x`
+func (v Version) HasBranchPrefix() bool {
+	return branchReleasePattern.MatchString(string(v))
+}
+
+// IsDev validates the value is a valid SemVer and does not container a "Prerelease" section
+func (v Version) IsDev() bool {
+	if v == "dev" {
+		return true
+	}
+
+	semVersion, err := semver.NewVersion(string(v))
+	return err != nil || // When version is not SemVer it is dev
+		semVersion.Prerelease() != "" // When the version includes pre-release assume dev
```

Comment:
Won't this make versions like `v2.12.1-alpha4` return `true` here and on `HasReleasePrefix`? Can that be a problem?